src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReduceToCollectionCommand.cs

```diff
@@ -10,19 +10,205 @@
 using Raven.Server.Utils;
 using Sparrow.Json;
 using Sparrow.Json.Parsing;
+using Sparrow.Platform;
 using Voron.Impl;
 
 namespace Raven.Server.Documents.Indexes.MapReduce.OutputToCollection
 {
-    public class OutputReduceToCollectionCommand : TransactionOperationsMerger.MergedTransactionCommand, IDisposable
+    public class OutputReduceToCollectionCommandBatcher : IDisposable
```

Comment:
The usual way we handle this is by having a single command that we enqueue multiple times.
Each time it runs, we process a batch, and the calling code will check if it is done and if not, resend to the merger.

Reply:
I'm not sure where this is handled like that? We are creating a lot of sub-commands all over the codebase.

Reply:
`ExecuteRateLimitedOperations` work in this manner.